src/System.Net.Http/tests/FunctionalTests/CustomContent.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            byte[] bytes = Encoding.UTF8.GetBytes(data);
+            Stream stream = new CustomStream(bytes, rewindable);
+            
+            return new CustomContent(stream, bytes.Length);
+        }
+
+        private CustomContent(Stream stream, long length) : base(stream)
+        {
+            _length = length;
+        }
+
+        public long Length
+        {
+            get
+            {
+                return _length;
+            }
+        }
+
+        private class CustomStream : Stream
+        {
+            private byte[] _buffer;
+            private long _position;
+            private bool _rewindable;
+
+            public CustomStream(byte[] buffer, bool rewindable)
+            {
+                _buffer = buffer;
+                _position = 0;
+                _rewindable = rewindable;
+            }
+        
+            public override bool CanRead
+            {
+                get { return true; }
+            }
+
+            public override bool CanSeek
+            {
+                get { return _rewindable; }
+            }
+
+            public override bool CanWrite
+            {
+                get { return false; }
+            }
+
+            public override void Flush()
+            {
+                throw new NotSupportedException();
+            }
+
+            public override long Length
+            {
+                get
+                {
+                    if (_rewindable)
+                    {
+                        return (long)_buffer.Length;
+                    }
+                    else
+                    {
+                        throw new NotSupportedException();
+                    }
+                }
+            }
+
+            public override long Position
+            {
+                get
+                {
+                    return _position;
+                }
+                set
+                {
+                    if (_rewindable)
+                    {
+                        _position = value;
+                    }
+                    else
+                    {
+                        throw new NotSupportedException();
+                    }
+                }
+            }
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                int bytesRead = 0;
+
+                for (int i = 0; i < count; i++)
+                {
+                    if (_position >= _buffer.Length)
+                    {
+                        break;
+                    }
+
+                    buffer[offset] = _buffer[_position];
+                    bytesRead++;
+                    offset++;
+                    _position++;
+                }
+
+                return bytesRead;
+            }
+
+            public override long Seek(long offset, SeekOrigin origin)
+            {
+                throw new NotImplementedException();
```

Comment:
Technically this should be implemented if CanSeek returns true.  Maybe this is just a moment-in-time thing?